pkg/artifactory/repositories.go

```diff
@@ -1271,199 +1267,52 @@ func isSelectRandom(opts ...bool) bool {
+	"pub":       {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"puppet":    {RepoLayoutRef: "puppet-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"pypi":      {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"sbt":       {RepoLayoutRef: "sbt-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"vagrant":   {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "federated": true}},
+	"vcs":       {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"remote": true}},
+	"rpm":       {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
 }
 
-//Return the default repo layout of Federated Repository type
-func getDefaultFederatedRepoLayoutRef(repoType string, opts ...bool) string {
-
-	selectRandom := isSelectRandom(opts...)
-	const defaultRepoLayout = "simple-default"
-	repoLayout := defaultRepoLayout
-
-	federatedDefaultRepoLayoutMap := map[string]string{
-		"alpine":    "simple-default",
-		"bower":     "bower-default",
-		"cran":      "simple-default",
-		"cargo":     "simple-default",
-		"chef":      "simple-default",
-		"cocoapods": "simple-default",
-		"composer":  "composer-default",
-		"conan":     "conan-default",
-		"conda":     "simple-default",
-		"debian":    "simple-default",
-		"docker":    "simple-default",
-		"gems":      "simple-default",
-		"generic":   "simple-default",
-		"gitlfs":    "simple-default",
-		"go":        "go-default",
-		"gradle":    "maven-2-default",
-		"helm":      "simple-default",
-		"ivy":       "ivy-default",
-		"maven":     "maven-2-default",
-		"npm":       "npm-default",
-		"nuget":     "nuget-default",
-		"opkg":      "simple-default",
-		"pub":       "simple-default",
-		"puppet":    "puppet-default",
-		"pypi":      "simple-default",
-		"sbt":       "sbt-default",
-		"vagrant":   "simple-default",
-		"rpm":       "simple-default",
-	}
-
-	if selectRandom {
-		repoLayout = selectRandomFromMapOfStrings(federatedDefaultRepoLayoutMap)
+//Return the default repo layout by Repository Type & Package Type
+func getDefaultRepoLayoutRef(repositoryType string, packageType string) string {
+	if defaultRepoLayoutMap[packageType].SupportedRepoTypes[repositoryType] == true {
+		return defaultRepoLayoutMap[packageType].RepoLayoutRef
 	} else {
-		if value, ok := federatedDefaultRepoLayoutMap[repoType]; ok {
-			repoLayout = value
-		}
+		panic(fmt.Errorf("Default Repo Layout not found for repository type: %s & package type: %s ", repositoryType, packageType))
```

Comment:
Please don't `panic` here. The func should return this as error so TF can handle it appropriately.

Take a look at the change @chb0github suggested here: https://github.com/jfrog/terraform-provider-artifactory/pull/335/files/886db2494e2d43ea749df6d132792b09b111c804..8b86d8aa22882e8c47f1271bbf2eb3c1f56d4f14#r814977949

Reply:
Implemented panic as per the suggestion https://github.com/jfrog/terraform-provider-artifactory/pull/335#discussion_r814986324

Reply:
Improved as suggested. Introduced error. TF will handle it as required.
```
func getDefaultRepoLayoutRef(repositoryType string, packageType string) (string, error)
```